src/shared/dissect-image.c

```diff
@@ -513,13 +513,10 @@ static int dissect_image(
                         m->encrypted = streq_ptr(fstype, "crypto_LUKS");
 
                         m->has_verity = verity && verity->data_path;
-                        m->verity_ready = m->has_verity &&
-                                verity->root_hash &&
-                                (verity->designator < 0 || verity->designator == PARTITION_ROOT);
+                        m->verity_ready = verity_settings_data_covers(verity, PARTITION_ROOT);
```

Comment:
this is no longer using `m->has_verity` is that intended?

Reply:
`m->has_verity` is initialized immediately before to `verity && verity->data_path`, and such a check *is* included in verity_settings_data_covers(), hence we don#t have to repeat the check here.

behaviour should be identical to status quo ante hence.